File: WoofAndroid/testutils/src/main/kotlin/com/perrystreet/woof/testutils/scheduler/TestSchedulerProvider.kt
package com.perrystreet.woof.testutils.scheduler

import com.perrystreet.woof.utils.scheduler.ISchedulerProvider
import io.reactivex.rxjava3.core.Scheduler
import io.reactivex.rxjava3.schedulers.Schedulers
import io.reactivex.rxjava3.schedulers.TestScheduler

class TestSchedulerProvider : ISchedulerProvider {
    val testScheduler = TestScheduler()

    override var mainScheduler: Scheduler = Schedulers.trampoline()
    override var ioScheduler: Scheduler = Schedulers.trampoline()
    override var computationScheduler: Scheduler = Schedulers.trampoline()

    fun useTestSchedulers() {
        mainScheduler = testScheduler
        ioScheduler = testScheduler
        computationScheduler = testScheduler
    }
}

File: WoofIOS/SwiftPackages/TestUtils/Sources/TestUtils/Scheduler/TestSchedulerProvider.swift
import CombineSchedulers
import Foundation
import Utils

public final class TestSchedulerProvider: SchedulerProviding {
    public let testScheduler = DispatchQueue.test
    public var mainScheduler: AnySchedulerOf<DispatchQueue> = .immediate
    public var ioScheduler: AnySchedulerOf<DispatchQueue> = .immediate
    public var computationScheduler: AnySchedulerOf<DispatchQueue> = .immediate

    public init() {}

    public func useTestSchedulers() {
        mainScheduler = testScheduler.eraseToAnyScheduler()
        ioScheduler = testScheduler.eraseToAnyScheduler()
        computationScheduler = testScheduler.eraseToAnyScheduler()
    }
}
